Extension/src/LanguageServer/settings.ts

```diff
@@ -414,337 +546,96 @@ export class CppSettings extends Settings {
-    }
-
-    public get vcFormatSpaceRemoveAroundUnaryOperator(): boolean {
-        return super.Section.get<boolean>("vcFormat.space.removeAroundUnaryOperator") === true;
-    }
-
-    public get vcFormatSpaceAroundBinaryOperator(): string {
-        // These strings have default values in package.json, so should never be undefined.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return super.Section.get<string>("vcFormat.space.aroundBinaryOperator")!;
-    }
-
-    public get vcFormatSpaceAroundAssignmentOperator(): string {
-        // These strings have default values in package.json, so should never be undefined.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return super.Section.get<string>("vcFormat.space.aroundAssignmentOperator")!;
-    }
-
-    public get vcFormatSpacePointerReferenceAlignment(): string {
-        // These strings have default values in package.json, so should never be undefined.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return super.Section.get<string>("vcFormat.space.pointerReferenceAlignment")!;
-    }
-
-    public get vcFormatSpaceAroundTernaryOperator(): string {
-        // These strings have default values in package.json, so should never be undefined.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return super.Section.get<string>("vcFormat.space.aroundTernaryOperator")!;
-    }
-
-    public get vcFormatWrapPreserveBlocks(): string {
-        // These strings have default values in package.json, so should never be undefined.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return super.Section.get<string>("vcFormat.wrap.preserveBlocks")!;
-    }
-
-    public get dimInactiveRegions(): boolean {
-        return super.Section.get<boolean>("dimInactiveRegions") === true
-            && this.intelliSenseEngine === "default"
-            && vscode.workspace.getConfiguration("workbench").get<string>("colorTheme") !== "Default High Contrast";
-    }
-
-    public get sshTargetsView(): string {
-        return super.Section.get<string>("sshTargetsView") ?? 'default';
-    }
+    public get defaultCompilerArgs(): string[] { return this.getAsArrayOfStrings("default.compilerArgs"); }
+    public get defaultCStandard(): string { return this.getAsString("default.cStandard"); }
+    public get defaultCppStandard(): string { return this.getAsString("default.cppStandard"); }
+    public get defaultConfigurationProvider(): string { return this.getAsString("default.configurationProvider"); }
+    public get defaultMergeConfigurations(): boolean { return this.getAsBoolean("default.mergeConfigurations"); }
+    public get defaultBrowsePath(): string[] { return this.getAsArrayOfStrings("default.browse.path", true); }
+    public get defaultDatabaseFilename(): string { return this.getAsString("default.browse.databaseFilename"); }
```

Comment:
This one needs the `?? undefined` put back in.  This setting is `string[] | undefined`

Reply:
I'll make one exception for a change to package.json for this case.  This setting is incorrectly missing "null" as one of the types in package.json even though the default is `null`.  You can add "null" to the list of types for this setting in package.json.

Reply:
Though I wonder if the reason it's not listed is because I don't know if the schema validation would be able to handle:

```
  "type": [
    "array",
    "null"
  ]
```

Reply:
I added the oneOf option to the default.browsePath setting.